### ISSUE TITLE
Development Code Fragment feature is not working

### ISSUE THREAD

**Issue by chrisyue** [phpmd/phpmd#345]
```
$ phpmd --version
PHPMD 2.3.2

$ phpmd path/to/File.php text design
```

the exit code is 0, but the File.php does contain `var_dump` and `print_r`

and the other features in design are working i think (I only test checking `die/exit`, `goto`, `eval`)

**Comment by manuelpichler** [comment]
Can you please provide a reproducable file for this issue

**Comment by chrisyue** [comment]
@manuelpichler here is the file content:

```php
<?php

namespace AppBundle\Controller\Front;

use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
use Symfony\Bundle\FrameworkBundle\Controller\Controller;
use Symfony\Component\HttpFoundation\Request;

/**
 * @Route("/hotsale")
 */
class HotSaleController extends Controller
{
    /**
     * @Route("/", name="front_hotsale_index", methods={"GET"})
     */
    public function indexAction(Request $request)
    {
        $hotsales = $this->getDoctrine()->getRepository('AppBundle:HotSale')
            ->createQueryBuilder('h')
            ->select('h, v')
            ->join('h.video', 'v')
            ->orderBy('h.sequence')
            ->getQuery()
            ->getResult();

        var_dump($hotsales);

        $banners = $this->get('app.banner.file_manager')->findAll();

        return $this->render('front/hotsale/index.html.twig', compact('hotsales', 'banners'));
    }
}
```

and I also found it maybe related to the `namespace`, because it can work after I remove the line of `namespace` declaration.

**Comment by chrisyue** [comment]
Maybe PHPMD thinks  `var_dump` is  `AppBundle\Controller\Front\var_dump`?

**Comment by ravage84** [comment]
@chrisyue  can you reduce the example to the following and still confirm the issue?

```` php
<?php

namespace AppBundle\Controller\Front;

class Foo
{
    public function testFoo()
    {
        var_dump($hotsales = '');
    }
}
````

**Comment by chrisyue** [comment]
@ravage84 same issue, won't work until I remove `namespace`.

And it is also strange that it will show the message `The method Foo::testFoo() calls the typical debug function var_dump() which is mostly only used during development.` twice, it should show only once? Maybe it's another issue, I don't know..

**Comment by torkiljohnsen** [comment]
I can confirm this issue.

**Comment by manuelpichler** [comment]
Works in the latest version that will be release today.

**Comment by chilion** [comment]
Any clue when this will be fixed? Our builds are not failing while they should now :octocat:

**Comment by ravage84** [comment]
@chilion have you updated PHPMD and PDepend to the newest version? With which version of these tools are you seeing this issue, still?

**Comment by chilion** [comment]
@ravage84 phpmd version 2.6.0 && pdepend version 2.5.0

**Comment by qkdreyer** [comment]
Still an issue for me :

![image](https://user-images.githubusercontent.com/717869/33612001-8ce51d96-d9d0-11e7-852d-61d6a8f18a00.png)

**Comment by stepansib** [comment]
Same for me( 
phpmd 2.6.0

**Comment by Barakur** [comment]
To add to this, because I noticed that too:
A `print_r` is not detected, but a `\print_r` is. So I guess https://github.com/phpmd/phpmd/issues/345#issuecomment-199611801 is on the right track.

**Comment by AJenbo** [comment]
Unable to reproduce, probably solved somewhere along the road.
